software/control/filterwheel.py

```diff
@@ -0,0 +1,64 @@
+import os
+import time
+
+from control._def import *
+
+class SquidFilterWheelWrapper:
+
+    def __init__(self, navigationController = None):
+        # emission filter position
+        self.w_pos_index = 0
+
+        self.navigationController = navigationController
+
+    def homing(self):
+        # make sure the start homing position is not at homing position  
+        starttime = time.time() 
+        self.navigationController.home_w()
+        self.navigationController.wait_till_operation_is_completed()
+        # judge whether the current position is a homing position or not
+        if time.time() - starttime > 0.02:
```

Comment:
Do we need this check?  Or if we always want the axis at `SQUID_FILTERWHEEL_OFFSET` after homing, can we just send it there no matter what?

Or is the issue that `home_w` may be a no-op if we're already homed?  In that case, would it make sense to check that the current W axis position is almost 0 (the home position) and if so, move to the offset?  That way at least we're more directly checking what we care about (the position) instead of just a sorta random time.

Reply:
@veerwang

Reply:
Let me do some test then answer this question. In my mind, when squid filter wheel is at the offset position then the function wait_till_operation_is_completed() would return quickly, so in this case we don't need keep the wheel moving to offset position again. we have not send the w axis position value to PC yet, so for easy reason, after doing some test I just do the implement of waiting some short time.